src/app/components/sidenav/sidenav.component.ts

```diff
@@ -11,7 +13,23 @@ export class SidenavComponent {
 
     @Output() readonly isSidenavOpenedChange = new EventEmitter<boolean>();
 
+    readonly basket: Product[] = [];
+
+    constructor(readonly productsService: ProductsService) {}
+
     toggleSidenavOpened() {
         this.isSidenavOpenedChange.emit(!this.isSidenavOpened);
     }
+
+    onItemBtnBuyClick(productId: string) {
+        const product = this.productsService.getById(productId);
+
+        if (product) {
+            this.basket.push(product);
+
+            if (!this.isSidenavOpened) {
+                this.toggleSidenavOpened();
+            }
+        }
```

Comment:
Рекомендую избегать излишней вложености за счет раних ретернов, это упрощает чтение и восприятие кода. Вот как можно было бы переписать код:
```suggestion
        if (!product) {
            return;
        }
        
        this.basket.push(product);
        
        if (!this.isSidenavOpened) {
            this.toggleSidenavOpened();
        }
```